### PR TITLE
allow alerts above or below the comparison line

### PR DESCRIPTION
**Pull request by eileenprice**

### CODE REVIEW

**Comment by Jbcampbe** [comment]
Not a huge fan of having a bunch of magic strings sitting around. Maybe we could get rid of/minimize those?

File: addon/components/column-chart/component.js [hunk 1]
```diff
@@ -179,14 +180,21 @@ export default BaseChartComponent.extend({
...
-                        return d3.select(this).attr('fill') === `url(#diagonalHatch${series.replaceIndex})`
-                        && d.data.value < comparisonValue;
+                        if (alert === 'below') {
+                            return d3.select(this).attr('fill') === `url(#diagonalHatch${series.replaceIndex})`
+                            && d.data.value < comparisonValue;
+                        }
+                        if (alert === 'above') {
+                            return d3.select(this).attr('fill') === `url(#diagonalHatch${series.replaceIndex})`
+                            && d.data.value > comparisonValue;
+                        }
```
Comment (Jbcampbe) [inline addon/components/column-chart/component.js]: ^ this could be cleaned up a little bit to get rid of duplication

**Comment by eileenprice** [comment]
I don't love the magic strings either, but I don't know how else to specify above/below, since it can't be both. There needs to exist "is there an alert?" (which could be a boolean) as well as "is the alert above or below the line?" I don't want to have two different booleans for this (e.g., `alertAbove` and `alertBelow`, because if they are both `true`, there will be unexpected results. And if they are both `false` but the `alert` variable is true, the results will also be unexpected. Do you have any ideas?

**Comment by eileenprice** [comment]
Also, I know I could have an `alert` boolean and an `aboveLine` boolean, where by default the alert is below the line but if `aboveLine` is true it will be above, but that seems clunky to me as well. Why is below the default? And it doesn't seem very intuitive to the consumer.

**Comment by Jbcampbe** [comment]
yeah I get what you're saying. We could just create an alert type enum

```
const AlertType = {
  ABOVE: 'above',
  BELOW: 'below'
}
```

makes the code a little bit easier to read and maintain

**Comment by eileenprice** [comment]
I don't quite understand (I don't have much experience with enums, and I didn't think they were possible in JavaScript). Wouldn't the template still have to pass one of the 'magic strings' "above" or "below" for that to work?

**Comment by JackieChiles** [comment]
> I don't quite understand (I don't have much experience with enums, and I didn't think they were possible in JavaScript). Wouldn't the template still have to pass one of the 'magic strings' "above" or "below" for that to work?

@eileenprice You are right that there are no true enums in javascript. That's part of the larger problem of JS not having a real type system. But, we try to use constants wherever possible across all languages to reduce the number of places these strings are defined, and especially to avoid using them as inline literals which are more vulnerable to typos.

**Pull request merged**